src/lib/Invitation/MailSender.php

```diff
@@ -21,49 +24,37 @@ final class MailSender implements InvitationSender
 
     private ConfigResolverInterface $configResolver;
 
-    private Swift_Mailer $mailer;
+    private NotificationServiceInterface $notificationService;
 
     public function __construct(
         Environment $twig,
         ConfigResolverInterface $configResolver,
-        Swift_Mailer $mailer
+        NotificationServiceInterface $notificationService
     ) {
         $this->twig = $twig;
         $this->configResolver = $configResolver;
-        $this->mailer = $mailer;
+        $this->notificationService = $notificationService;
     }
 
     public function sendInvitation(Invitation $invitation): void
     {
-        $template = $this->twig->load(
-            $this->configResolver->getParameter(
-                'user_invitation.templates.mail',
-                null,
-                $invitation->getSiteAccessIdentifier()
-            )
-        );
-
-        $senderAddress = $this->configResolver->hasParameter('sender_address', 'swiftmailer.mailer')
-            ? $this->configResolver->getParameter('sender_address', 'swiftmailer.mailer')
-            : '';
-
-        $subject = $template->renderBlock('subject', []);
-        $from = $template->renderBlock('from', []) ?: $senderAddress;
-        $body = $template->renderBlock('body', [
-            'invite_hash' => $invitation->getHash(),
-            'siteaccess' => $invitation->getSiteAccessIdentifier(),
-            'invitation' => $invitation,
-        ]);
+        if (!$this->isNotifierConfigured()) {
+            return;
+        }
 
-        $message = (new Swift_Message())
-            ->setSubject($subject)
-            ->setTo($invitation->getEmail())
-            ->setBody($body, 'text/html');
+        $this->notificationService->send(
+            new SymfonyNotificationAdapter(
+                new UserInvitation($invitation, $this->configResolver, $this->twig),
+            ),
```

Comment:
In the future this would probably be a good location for introducing a Factory - to allow changing how notifications are built, if a developer wishes to.

Reply:
was thinkg about it tbh, but didnt want to over engineer it at the start.